app/src/main/java/org/simple/clinic/instantsearch/InstantSearchScreen.kt

```diff
@@ -199,6 +201,11 @@ class InstantSearchScreen :
     searchResultsAdapter.addLoadStateListener(::searchResultsAdapterLoadStateListener)
   }
 
+  override fun onStart() {
+    super.onStart()
+    lifecycleEvents.onNext(InstantSearchScreenShown)
```

Comment:
~Why did we choose onstart, when the same thing was happening in Init?~ Okay so first we wanna load the facility and then check the prefill and keyboard, is that right?

Reply:
Yeah, partly. But more importantly it has to do with lifecycle of Fragments and ViewModel. Fragments have a tricky lifecycle. Here we are trying to do certain actions when an instant search screen is shown (showing keyboard and prefilling search). 

When we are opening a search result we are simply destroying the view in the fragment and not the entire fragments, so viewmodel is not cleared. So when we revisit the search screen the keyboard and prefill doesn't work if these are present in `Init`.

For that reason since these actions are tied to lifecycle of fragment rather than ViewModel, we are sending a UiEvent to do that action everytime the instant search screen is started.

This article explains the lifecycles well : https://zsmb.co/fragment-lifecycles-in-the-age-of-jetpack/

Reply:
Ah now it makes much more sense.